### ISSUE TITLE
Subdomain localized urls

### ISSUE THREAD

**Issue by jeffsmor** [codezero-be/laravel-localized-routes#73]
Hi...

I have the folowing supported locale array:

```php
 'supported-locales' => [ 
        'en',
        'pt',
        'es',
    ],
```

And the following home blade view:

```php
    {{ Route::localizedUrl('en') }}<br>
    {{ Route::localizedUrl('pt') }}<br>
    {{ Route::localizedUrl('es') }}
```

When I get mysite.test I see:

```
https://mysite.test
https://mysite.test/pt
https://mysite.test/es
```

Everything ok, when I change my supported locale array to:

```php
    'supported-locales' => [
        'en' => 'mysite.test',
        'pt' => 'pt.mysite.test',
        'es' => 'es.mysite.test',
    ],
```

And get mysite.test again I see:

```
https://video-site.test
https://video-site.test
https://video-site.test
```

Is it a package issue?

Amazing package, by the way...

**Comment by ivanvermeyen** [comment]
Hello,

Did you call `Route::localizedUrl('en')` on a localized route?

If you call it on a localized route, then it should work.

However, if you call it on a route that is registered outside the `Route::localized()` closure, then it returns the same URL.
This is because there are no localized versions of that route.

I did forget to include a test for that, so I'll add it.

**Comment by jeffsmor** [comment]
> Hello,
> 
> Did you call `Route::localizedUrl('en')` on a localized route?
> 
> If you call it on a localized route, then it should work.
> 
> However, if you call it on a route that is registered outside the `Route::localized()` closure, then it returns the same URL. This is because there are no localized versions of that route.
> 
> I did forget to include a test for that, so I'll add it.

Yes.

This is my routes/web.php

````php
Route::localized(function () {
    Route::get('/', function (Request $request) {
        return view('home');
    })->name('home');
});
````

And I'm calling the `Route::localizedUrl('en')` inside the view home.

**Comment by ivanvermeyen** [comment]
According to my test this should work.
I also copied your setup to a fresh Laravel project and put it on GitHub.
https://github.com/ivanvermeyen/domain-test

If you open the homepage at domain-test.test (the domains I configured) you should see 3 URL's:

<img width="517" alt="Scherm­afbeelding 2023-02-19 om 13 15 06" src="https://user-images.githubusercontent.com/3598622/219947312-1e4828d0-dd78-4d37-abc0-c845d49f3835.png">

I've added you to the project, so if you can replicate the issue I can try to fix it 👍

**Comment by jeffsmor** [comment]
Hi man,,,

I got your project and it is really functional...

I came back to my one and I found out the issue.

I set:

````php
'fallback_locale' => 'en',
````

When it is `null` I have what I am expecting.

However, I am just at the beggining of my project and I will planning soon how the routes localized will work.

Thanks for now.

**Comment by ivanvermeyen** [comment]
That's definitely not working as intended :)
I see it too now.
Thanks for catching this, I will look into it.

**Comment by ivanvermeyen** [comment]
Should be fixed in 2.9.1 👍
